lib/VisualBuilder.js

```diff
@@ -211,9 +211,43 @@ class VisualBuilder extends EventEmitter {
             catch (e) {
                 return reject('Failed updating visual capabilities');
             }
+
+            //load and validate dependencies if the file exists for the visual
+            let loadDependencies = false;
+            let dependenciesFilePath = this.visualPackage.buildPath(visualConfig.dependencies);
```

Comment:
We should check if `visualConfig.dependencies` is defined and and skip the rest of this if it is not. Visuals created with other templates or before this was added will have this undefined so we should check if it's defined.